internal/cli/performanceadvisor/slowquerylogs/list.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package slowquerylogs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const listTemplate = `NAMESPACE{{range .SlowQuery}}
+{{.Namespace}}{{end}}
+`
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	store       store.PerformanceAdvisorSlowQueriesLister
+	processName string
+	hostID      string
+	since       int64
+	duration    int64
+	namespaces  string
+	nLog        int64
+}
+
+func (opts *ListOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *ListOpts) Run() error {
+	host, err := opts.host()
+	if err != nil {
+		return err
+	}
+	r, err := opts.store.PerformanceAdvisorSlowQueries(opts.ConfigProjectID(), host, opts.newSlowQueryOptions())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *ListOpts) newSlowQueryOptions() *atlas.SlowQueryOptions {
+	return &atlas.SlowQueryOptions{
+		Namespaces: opts.namespaces,
+		NLogs:      opts.nLog,
+		NamespaceOptions: atlas.NamespaceOptions{
+			Since:    opts.since,
+			Duration: opts.duration,
+		},
+	}
+}
+
+func (opts *ListOpts) validateProcessName() error {
+	const length = 2
+	process := strings.Split(opts.processName, ":")
+	if len(process) != length {
+		return fmt.Errorf("'%v' is not valid", opts.processName)
+	}
+	return nil
+}
+
+func (opts *ListOpts) host() (string, error) {
+	if opts.processName == "" {
+		return opts.hostID, nil
+	}
+	err := opts.validateProcessName()
+	if err != nil {
+		return "", err
+	}
+	return opts.processName, nil
+}
+
+func (opts *ListOpts) markRequired(cmd *cobra.Command) func() error {
+	return func() error {
+		if config.Service() == config.CloudService {
+			return cmd.MarkFlagRequired(flag.ProcessName)
+		}
+		return cmd.MarkFlagRequired(flag.HostID)
+	}
+}
```

Comment:
[nit] it feels like these will be copy/pasted again for the next command, we should extract them to a shared space